### ISSUE TITLE
[Feat] Reproducibility

### ISSUE THREAD

**Issue by bcebere** [vanderschaarlab/temporai#19]
## Feature Description
Every plugin/pipeline or API call should support fixing the random seed.

There are methods for setting a global random seed. For example
```python
# stdlib
import random

# third party
import numpy as np
import torch


def enable_reproducible_results(random_state: int = 0) -> None:
    np.random.seed(random_state)
    torch.manual_seed(random_state)
    random.seed(random_state)
```
